File: MoonEngine-Root/ECS/src/main/java/com/greentree/engine/moon/ecs/pool/EmptyEntityStrategy.java
package com.greentree.engine.moon.ecs.pool;

import java.util.Comparator;

import com.greentree.engine.moon.ecs.Entity;

public class EmptyEntityStrategy implements EntityPoolStrategy {
	private static final long serialVersionUID = 1L;
	
	private static final Comparator<Entity> COMPARATOR = Comparator.comparing(e -> e.size());

	@Override
	public Comparator<Entity> comporator() {
		return COMPARATOR;
	}

	@Override
	public void toInstance(Entity entity) {
		entity.clear();
	}

}

File: MoonEngine/ECS/src/main/java/com/greentree/engine/moon/ecs/pool/EmptyEntityStrategy.kt
package com.greentree.engine.moon.ecs.pool

import com.greentree.engine.moon.ecs.Entity

object EmptyEntityStrategy : EntityPoolStrategy {

	override fun comporator(): Comparator<Entity> {
		return COMPARATOR
	}

	override fun toInstance(entity: Entity) {
		entity.clear()
	}

	private val COMPARATOR = Comparator.comparing { e: Entity -> e.size() }
}